### PR TITLE
[BUG] Pass parquet2 io errors correctly into arrow2

### PR DESCRIPTION
**Pull request by desmondcheongzx**
In https://github.com/Eventual-Inc/Daft/commit/7fe3dbceb3ca3ee0c555753f0a55bd7e0c57bba6, two changes were made that caused parquet2 io errors to be mishandled.

Firstly, a `IoError` for parquet2 was introduced.  However in the implementation of `parquet2::error::Error` for `arrow2::error:Error`, any parquet2 error that is not `FeatureNotActive` or `Transport` is transformed into an `ExternalFormat` error.

Secondly, arrow2 IO errors are intended to be marked as `ByteStreamError`s, primarily so that they are handled as transient errors and can be retried appropriately. However this special case was removed.

This PR makes now classifies parquet2 io errors as arrow2 io errors, and arrow2 io errors are now marked as `ByteStreeamError`s.

### CODE REVIEW

File: src/arrow2/src/io/parquet/mod.rs [hunk 1]
```diff
@@ -22,6 +22,9 @@ impl From<parquet2::error::Error> for Error {
             parquet2::error::Error::Transport(msg) => {
                 Error::Io(std::io::Error::new(std::io::ErrorKind::Other, msg))
             }
+            parquet2::error::Error::IoError(msg) => {
+                Error::Io(std::io::Error::new(std::io::ErrorKind::Other, msg))
+            }
```
Comment (jaychia) [inline src/arrow2/src/io/parquet/mod.rs:27]: makes sense to me... Will let @samster25 do a sign-off

Comment (samster25) [reply src/arrow2/src/io/parquet/mod.rs:27]: this looks good.

File: src/common/error/src/error.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@ pub enum DaftError {
     #[error("DaftError::ComputeError {0}")]
     ComputeError(String),
     #[error("DaftError::ArrowError {0}")]
-    ArrowError(#[from] arrow2::error::Error),
+    ArrowError(String),
```
Comment (samster25) [inline src/common/error/src/error.rs:17]: This can still hold `arrow2::error::Error` instead of a string

**Review by samster25 — APPROVED** [review]

File: src/arrow2/src/io/parquet/mod.rs (hunk 1 above)
Comment (raunakab) [reply src/arrow2/src/io/parquet/mod.rs:27]: Minor nit; could instead be:
```rs
[par](parquet2::error::Error::Transport(msg)) | parquet2::error::Error::IoError(msg) => ...
```

But I mean doesn't matter too much. I'm good with current.

**Review by raunakab — APPROVED** [review]

**Comment by raunakab** [comment]
Sorry for the regression, folks. I must have accidentally removed a `From` implementation that shouldn't have been removed.

File: src/common/error/src/error.rs (hunk 2 above)
Comment (desmondcheongzx) [reply src/common/error/src/error.rs:17]: Ah gotcha, done

File: src/arrow2/src/io/parquet/mod.rs (hunk 1 above)
Comment (desmondcheongzx) [reply src/arrow2/src/io/parquet/mod.rs:27]: unfortunately one of the `msg` is `String`, and the other is an `std::io::Error` :')

**Pull request merged**